### PR TITLE
Extract elastic layout properties to a separate redux slice

### PR DESCRIPTION
**Pull request by cbeer**
Both @jkeck and I have been... surprised... about the width/height properties of window, when they only apply to elastic mode. Hopefully moving them to a new slice will make this a little more approachable.

### CODE REVIEW

File: src/state/reducers/elasticLayout.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import { updateIn, merge } from 'immutable';
+import ActionTypes from '../actions/action-types';
+
+/**
+ * windowsReducer
```
Comment (mejackreed) [inline src/state/reducers/elasticLayout.js]: `elasticLayoutReducer`?

Comment (cbeer) [reply src/state/reducers/elasticLayout.js]: 👍

File: __tests__/src/actions/window.test.js [hunk 2]
```diff
@@ -12,14 +12,18 @@ describe('window actions', () => {
 
       const mockState = {
         companionWindows: {},
-        windows: {
+        elasticLayout: {
           window: {
```
Comment (mejackreed) [inline __tests__/src/actions/window.test.js]: Should there be another `windows` layer here?

Comment (cbeer) [reply __tests__/src/actions/window.test.js]: I don't think so; `window` is just a terrible name for the window id there.

**Comment by cbeer** [comment]
You're right; I was missing a commit from #2454. Juggling branches gets confusing.

**Pull request closed without merging**